src/mono/wasm/debugger/BrowserDebugProxy/DevToolsHelper.cs

```diff
@@ -502,4 +506,69 @@ public PerScopeCache()
         {
         }
     }
+
+    internal sealed class ConcurrentExecutionContextDictionary
+    {
+        private ConcurrentDictionary<SessionId, ConcurrentBag<ExecutionContext>> contexts = new ();
+        public ExecutionContext GetCurrentContext(SessionId sessionId)
+            => TryGetCurrentExecutionContextValue(sessionId, out ExecutionContext context)
+                ? context
+                : throw new KeyNotFoundException($"No execution context found for session {sessionId}");
+
+        public bool TryGetCurrentExecutionContextValue(SessionId id, out ExecutionContext executionContext)
+        {
+            executionContext = null;
+            if (!contexts.TryGetValue(id, out ConcurrentBag<ExecutionContext> contextBag))
+                return false;
+            if (contextBag.IsEmpty)
+                return false;
+            executionContext = contextBag.Where(context => context.Id == contextBag.Where(context => context.Destroyed == false).Max(context => context.Id)).FirstOrDefault();
```

Comment:
This should work, right? Fine for a follow up PR.
```suggestion
            int maxId = contextBag.Where(context => context.Destroyed == false).Max(context => context.Id);
            executionContext = contextBag.Where(context => context.Id == maxId).FirstOrDefault();
```